Formula/git-when-merged.rb

```diff
@@ -1,21 +1,17 @@
 class GitWhenMerged < Formula
-  include Language::Python::Shebang
+  include Language::Python::Virtualenv
 
   desc "Find where a commit was merged in git"
-  homepage "https://github.com/mhagger/git-when-merged"
-  url "https://github.com/mhagger/git-when-merged/archive/v1.2.1.tar.gz"
-  sha256 "46ba5076981862ac2ad0fa0a94b9a5401ef6b5c5b0506c6e306b76e5798e1f58"
+  homepage "https://pypi.org/project/git-when-merged/"
+  url "https://files.pythonhosted.org/packages/35/45/a2696a3a133838739baaa20fbb180b98c2c053970b2723b4384cbbc1f4a0/git-when-merged-1.2.1.tar.gz"
```

Comment:
What's the advantage of using the PyPI source?

Reply:
while git-when-merged is a simple single file application currently, there is no meaningful difference, but if it has evolved into an app with multi dependencies, installing from its python package is the most convenient option.

Reply:
Are upstream looking likely to make that change in the near future?

Using `pip` produces bottles with different checksums across OSes/macOS versions (so prevents the creation of an `:all` bottle) and unnecessarily forces usage of a particular `python3`.

We currently do the latter at the moment, but that can (and should) be fixed.

Reply:
Also, won't the hash in the url breaks automatic bump through `brew bump` and `brew bump-formula-pr`?

Reply:
> Are upstream looking likely to make that change in the near future?

I am not sure, but if upstream published a new release, PyPI is a canonical source and be updated, no need to watch the source github repo.

> We currently do the latter at the moment, but that can (and should) be fixed.

anything I can do here?

Reply:
> Also, won't the hash in the url breaks automatic bump through `brew bump` and `brew bump-formula-pr`?

all python virtualenv based formula works this way, I guess it is fine.

Reply:
> > We currently do the latter at the moment, but that can (and should) be fixed.
> 
> anything I can do here?

Thanks for asking, but I'm already fixing this at #107905.

Reply:
> Also, won't the hash in the url breaks automatic bump through `brew bump` and `brew bump-formula-pr`?

I don't think it will, actually. `brew bump-formula-pr` knows how to update PyPI `url`s.